lib/hypothesis/engine.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Hypothesis
+  class Engine
+    attr_reader :current_source
+
+    def initialize(options)
+      @max_examples = options.fetch(:max_examples, 200)
```

Comment:
Would it make sense to explicitly list keyword parameters in the declaration of`initialize`? Might make it clearer at a glance which options are legal, and would also serve to reject invalid options:

```ruby
def initialize(max_examples:, seed:)
  # ...
end
```

Reply:
Might do! How does that interact with just wanting to do pass-through from the hypothesis method? (Proper keyword arguments happened after I last used Ruby, so I'm not that familiar with the feature)

Reply:
The interaction is pretty ergonomic and looks a lot like Python. Passing through keyword arguments might look like

```ruby
def wrapped_method(keyword_one:, keyword_two:, etc:)
  # ...
end

def wrapper_method(**keywords_to_pass_through)
  # keywords_to_pass_through is a hash
  keyword_defaults = {keyword_one: 'value_one'}
  wrapped_method(
    **keyword_defaults.merge(keywords_to_pass_through)
  )
end
```